scripts/modules/config-manager.js

```diff
@@ -669,6 +669,34 @@ function isConfigFilePresent(explicitRoot = null) {
 	return fs.existsSync(configPath);
 }
 
+/**
+ * Gets the user ID from the configuration.
+ * @param {string|null} explicitRoot - Optional explicit path to the project root.
+ * @returns {string|null} The user ID or null if not found.
+ */
+function getUserId(explicitRoot = null) {
+	const config = getConfig(explicitRoot);
+	if (!config.global) {
+		config.global = {}; // Ensure global object exists
+	}
```

Comment:
shouldn't it be like a fail fast here ?

```suggestion
	if (!config.global) {
		return;
	}
```

Reply:
No this creates it if it doesn't exist so it can write to it below

Reply:
good one